src/solvers/dgsem_tree/dg_3d_compressible_euler.jl

```diff
@@ -262,3 +262,353 @@
+     StaticInt(nvariables(equations) + 2))) # We also compute "+ 2" logs
+
+  @turbo for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+    rho    = u_cons[1, i, j, k, element]
+    rho_v1 = u_cons[2, i, j, k, element]
+    rho_v2 = u_cons[3, i, j, k, element]
+    rho_v3 = u_cons[4, i, j, k, element]
+    rho_e  = u_cons[5, i, j, k, element]
+
+    v1 = rho_v1 / rho
+    v2 = rho_v2 / rho
+    v3 = rho_v3 / rho
+    p = (equations.gamma - 1) * (rho_e - 0.5 * (rho_v1 * v1 + rho_v2 * v2 + rho_v3 * v3))
+
+    u_prim[i, j, k, 1] = rho
+    u_prim[i, j, k, 2] = v1
+    u_prim[i, j, k, 3] = v2
+    u_prim[i, j, k, 4] = v3
+    u_prim[i, j, k, 5] = p
+    u_prim[i, j, k, 6] = log(rho)
+    u_prim[i, j, k, 7] = log(p)
+  end
+
+
+  # x direction
+  # At first, we create new temporary arrays with permuted memory layout to
+  # allow using SIMD instructions along the first dimension (which is contiguous
+  # in memory).
+  du_permuted = StrideArray{eltype(u_cons)}(undef,
+    (StaticInt(nnodes(dg)^2), StaticInt(nnodes(dg)),
+     StaticInt(nvariables(equations))))
+
+  u_prim_permuted = StrideArray{eltype(u_cons)}(undef,
+    (StaticInt(nnodes(dg)^2), StaticInt(nnodes(dg)),
+     StaticInt(nvariables(equations) + 2)))
+
+  @turbo for v in eachvariable(equations),
+             k in eachnode(dg),
+             j in eachnode(dg),
+             i in eachnode(dg)
+    jk = j + nnodes(dg) * (k- 1)
+    u_prim_permuted[jk, i, v] = u_prim[i, j, k, v]
+  end
+  fill!(du_permuted, zero(eltype(du_permuted)))
+
+  # Next, we basically inline the volume flux. To allow SIMD vectorization and
+  # still use the symmetry of the volume flux and the derivative matrix, we
+  # loop over the triangular part in an outer loop and use a plain inner loop.
+  for i in eachnode(dg), ii in (i+1):nnodes(dg)
+    @turbo for jk in Base.OneTo(nnodes(dg)^2)
+      rho_ll = u_prim_permuted[jk, i, 1]
+      v1_ll  = u_prim_permuted[jk, i, 2]
+      v2_ll  = u_prim_permuted[jk, i, 3]
+      v3_ll  = u_prim_permuted[jk, i, 4]
+      p_ll   = u_prim_permuted[jk, i, 5]
+      log_rho_ll = u_prim_permuted[jk, i, 6]
+      log_p_ll   = u_prim_permuted[jk, i, 7]
+
+      rho_rr = u_prim_permuted[jk, ii, 1]
+      v1_rr  = u_prim_permuted[jk, ii, 2]
+      v2_rr  = u_prim_permuted[jk, ii, 3]
+      v3_rr  = u_prim_permuted[jk, ii, 4]
+      p_rr   = u_prim_permuted[jk, ii, 5]
+      log_rho_rr = u_prim_permuted[jk, ii, 6]
+      log_p_rr   = u_prim_permuted[jk, ii, 7]
+
+      # Compute required mean values
+      # We inline the logarithmic mean to allow LoopVectorization.jl to optimize
+      # it efficiently. This is equivalent to
+      #   rho_mean = ln_mean(rho_ll, rho_rr)
+      x1 = rho_ll
+      log_x1 = log_rho_ll
+      y1 = rho_rr
+      log_y1 = log_rho_rr
+      x1_plus_y1 = x1 + y1
+      y1_minus_x1 = y1 - x1
+      z1 = y1_minus_x1^2 / x1_plus_y1^2
+      special_path1 = x1_plus_y1 / (2 + z1*(0.666666666666667 + z1*(0.4 + 0.285714285714286*z1)))
```

Comment:
Is this faster than
```suggestion
      special_path1 = x1_plus_y1 / (2 + z1*(2/3 + z1*(0.4 + 0.285714285714286*z1)))
```
?

Reply:
Improved in cfa2eff0e42db1b8c5144abe60c38c48e99b73e9